### PR TITLE
openstack: Make the config.networkV2Client conditional in StepAllocateIp

### PR DESCRIPTION
**Pull request by cb-oath**
In cases where the OpenStack environment does not contain a v2 network, the
builder will error out.  We only need the `networkV2Client` when using floating
IPs.  This will allow packer to work in environments where a v2 network is not
available.

### CODE REVIEW

File: builder/openstack/step_allocate_ip.go [hunk 1]
```diff
@@ -29,12 +30,15 @@ func (s *StepAllocateIp) Run(_ context.Context, state multistep.StateBag) multis
...
 
-	// We need the v2 network client
-	networkClient, err := config.networkV2Client()
-	if err != nil {
-		err = fmt.Errorf("Error initializing network client: %s", err)
-		state.Put("error", err)
-		return multistep.ActionHalt
+	// We might need the v2 network client
+	var networkClient *gophercloud.ServiceClient
+	if s.FloatingIP != "" || s.ReuseIPs || s.FloatingIPNetwork != "" {
```
Comment (rickard-von-essen) [inline builder/openstack/step_allocate_ip.go]: This code would be much cleaner it this is a guard clause.

File: builder/openstack/step_allocate_ip.go [hunk 2]
```diff
@@ -147,15 +151,15 @@ func (s *StepAllocateIp) Cleanup(state multistep.StateBag) {
...
 
-	// We need the v2 network client
-	client, err := config.networkV2Client()
-	if err != nil {
-		ui.Error(fmt.Sprintf(
-			"Error deleting temporary floating IP '%s' (%s)", instanceIP.ID, instanceIP.FloatingIP))
-		return
-	}
+	if instanceIP.FloatingIP != "" && instanceIP.ID != "" {
+		// We need the v2 network client
```
Comment (rickard-von-essen) [inline builder/openstack/step_allocate_ip.go]: And this.

**Comment by rickard-von-essen** [comment]
> This will allow packer to work in environments where a v2 network is not
available.

Could you explain when this is the case? This seems like a very special setup?

**Comment by cb-oath** [comment]
Thanks, just added guard clauses.  I had to move the instanceIP declaration up and store it in the state bag to avoid an error in ssh.go.  It also makes the teardown part of the patch cleaner.  

Yes.. unfortunately, we have an older OpenStack setup that does not run a V2 network and is a bit of a snowflake.  Hopefully we can work out a way to handle this cleanly so we don't need to maintain a packer fork! :)  Thanks for the help.

**Comment by SwampDragons** [comment]
@rickard-von-essen looks like this has been ready for a new review for a while; are you still feeling up for reviewing it?

**Comment by rickard-von-essen** [comment]
Thanks!

**Pull request merged**